File: typescript-examples/src/module1/testDoubles/mock/AccountService.ts
import {Logger} from "./Logger.ts";

export class AccountService {
    private readonly logger: Logger

    constructor(logger: Logger) {
        this.logger = logger;
    }

    public addDeposit(amount: number): void {
        // Here goes the "add deposit" logic

        this.logger.log("Deposit added: " + amount);
    }
}

File: src/main/java/com/codurance/module1/TestDoubles/Mock/AccountService.java
package com.codurance.module1.TestDoubles.Mock;

public class AccountService {
    private Logger logger;

    public AccountService(Logger logger) {
        this.logger = logger;
    }

    public void addDeposit(double amount) {
        // Here goes the "add deposit" logic

        logger.log("Deposit added: " + amount);
    }
}
